File: minting/src/main/scala/co/topl/minting/algebras/VrfCalculatorAlgebra.scala
package co.topl.minting.algebras

import co.topl.models._

trait VrfCalculatorAlgebra[F[_]] {

  def rhoForSlot(slot: Slot, eta: Eta): F[Rho]

  def proofForSlot(slot: Slot, eta: Eta): F[Bytes]

}

File: dart/minting/lib/algebras/vrf_calculator_algebra.dart
import 'package:fixnum/fixnum.dart';
import 'package:fpdart/fpdart.dart';
import 'package:rational/rational.dart';

abstract class VrfCalculatorAlgebra {
  Future<List<int>> rhoForSlot(Int64 slot, List<int> eta);
  Future<List<int>> proofForSlot(Int64 slot, List<int> eta);
  Future<List<Int64>> ineligibleSlots(Int64 epoch, List<int> eta,
      Tuple2<Int64, Int64>? slotRange, Rational relativeStake);
}
